scripts/utilities/supported_boards.py

```diff
@@ -1,19 +1,32 @@
-def get_board_names():
+import os.path
+from os import path, listdir
+
+
+def pathIsFolderAndHasIOCFile(pathName: str):
+    isFolder = not path.isfile(pathName)
+    if not isFolder:
+        return False
+    hasIOC = any(i.endswith(".ioc") for i in listdir(pathName)) or any(
+        [i.endswith(".ioc") for i in listdir(path.join(pathName, "src/cubemx"))])
+    return hasIOC
```

Comment:
A few nits: 
- `i` usually indicates index, but these are files. `file` would probably be a more descriptive name
- can you change casing in this file from `camelCase` to `snake_case`? It's what we've been going with and is the standard for Python